### PR TITLE
I2c navio bebop

### PR DESCRIPTION
**Pull request by lgywata**
More drivers using the new I2CDevice. I'm planing to test these drivers this week, but any feedback is well come. @lucasdemarchi

### CODE REVIEW

File: libraries/AP_HAL/I2CDevice.h [hunk 1]
```diff
@@ -63,6 +63,11 @@ class I2CDevice : public Device {
 
     /* See Device::get_fd() */
     virtual int get_fd() override = 0;
+
+    void ignore_errors(bool b) { _ignore_errors = b; }
+
+protected:
+    bool _ignore_errors;
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/I2CDevice.h]: if this interface is changed, it needs to be in a separate commit.

In this particular case it's a bad change since we do nothing with that info.

File: libraries/AP_Notify/NavioLED_I2C.cpp [hunk 2]
```diff
@@ -26,37 +26,33 @@ extern const AP_HAL::HAL& hal;
...
 
     // take i2c bus sempahore
-    if (!i2c_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+    if (!_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
         return false;
     }
 
     // disable recording of i2c lockup errors
-    hal.i2c->ignore_errors(true);
+    _dev->ignore_errors(true);
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/NavioLED_I2C.cpp]: just remove this. We never do anything with it.

File: libraries/AP_Notify/NavioLED_I2C.cpp [hunk 3]
```diff
@@ -26,37 +26,33 @@ extern const AP_HAL::HAL& hal;
 
 bool NavioLED_I2C::hw_init()
 {
-    // get pointer to i2c bus semaphore
-    AP_HAL::Semaphore* i2c_sem = hal.i2c->get_semaphore();
+    _dev = hal.i2c_mgr->get_device(0, PCA9685_ADDRESS);
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/NavioLED_I2C.cpp]: bus looks wrong... navio doesn't have bus 0. It's rather 1.

File: libraries/AP_Notify/NavioLED_I2C.h [hunk 4]
```diff
@@ -17,10 +17,14 @@
 #pragma once
 
 #include "NavioLED.h"
+#include <AP_HAL/I2CDevice.h>
 
 class NavioLED_I2C : public NavioLED
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/NavioLED_I2C.h]: Thinking again about this driver. I think we could just replace with our newer ones based on RCOutputRGBLed. They have the benefit that they integrate well with changes to the motors rather than interfering with them.

Comment (lucasdemarchi) [inline libraries/AP_Notify/NavioLED_I2C.h]: anyway... since the conversion is already done, we can get it in once the issues pointed above are fixed.

File: libraries/AP_HAL_Linux/RCOutput_Bebop.cpp [hunk 5]
```diff
@@ -134,12 +135,11 @@ bool RCOutput_Bebop::_get_info(struct bldc_info *info)
...
     }
     memset(info, 0, sizeof(struct bldc_info));
-    if (!_i2c_sem->take(0)) {
+    if (!_dev->get_semaphore()->take(0)) {
         return false;
     }
-    hal.i2c1->readRegisters(BEBOP_BLDC_I2C_ADDR, BEBOP_BLDC_GET_INFO,
-                            sizeof(struct bldc_info), (uint8_t *)info);
-    _i2c_sem->give();
+    _dev->read_registers(BEBOP_BLDC_GET_INFO, (uint8_t*)info, sizeof(info));
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/RCOutput_Bebop.cpp]: sizeof(info) is the size of a pointer to info.... you need to keep the sizeof as before

Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/RCOutput_Bebop.cpp]: or use sizeof(*info)

File: libraries/AP_HAL/I2CDevice.h (hunk 1 above)
Comment (lgywata) [inline libraries/AP_HAL/I2CDevice.h]: Ack.

File: libraries/AP_Notify/NavioLED_I2C.cpp (hunk 2 above)
Comment (lgywata) [inline libraries/AP_Notify/NavioLED_I2C.cpp]: Ack.

File: libraries/AP_Notify/NavioLED_I2C.cpp (hunk 3 above)
Comment (lgywata) [inline libraries/AP_Notify/NavioLED_I2C.cpp]: ok, thanks.

File: libraries/AP_HAL_Linux/RCOutput_Bebop.cpp (hunk 5 above)
Comment (lgywata) [inline libraries/AP_HAL_Linux/RCOutput_Bebop.cpp]: uh. Ack.

**Comment by lgywata** [comment]
see #3905
